### PR TITLE
Fix color picker sample skill

### PR DESCRIPTION
**Pull request by Chris-Liao**
## Description

Fix color picker sample, for null slot input and session ending process.

## Motivation and Context

Customer report issue about color picker, #157.

## Testing

Test in developer console and lambda.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

[issues]: https://github.com/alexa/alexa-skills-kit-sdk-for-java/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java [hunk 1]
```diff
@@ -35,18 +36,26 @@ public boolean canHandle(HandlerInput input) {
         String speechText;
         String favoriteColor = (String) input.getAttributesManager().getSessionAttributes().get(COLOR_KEY);
 
+        boolean isAskResponse = false;
```
Comment (tianrenz) [inline samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java]: Seems what this boolean flag does is the opposite of what the variable name means. 

If the response is a "Ask" response, we want to keep the session open (set shouldEndSession to false). L55 is doing the opposite. 

Also, what's the reason for moving this logic from ``MyColorIsIntentHandler`` to here?

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java [hunk 2]
```diff
@@ -30,7 +30,8 @@ public boolean canHandle(HandlerInput input) {
 
     @Override
     public Optional<Response> handle(HandlerInput input) {
-        String speechText = "Welcome to the Alexa Skills Kit sample. Please tell me your favorite color by saying, my favorite color is red";
+        String speechText = "Welcome to the Alexa Skills Kit sample. Please tell me your favorite color by saying, my favorite color is red. " +
```
Comment (tianrenz) [inline samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java]: Are those utterance changes reviewed by COPs?

Comment (Chris-Liao) [reply samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java]: @tianrenz , are you asking adding `"Your options are green, blue, purple, red, orange and yellow."`? This [ticket](https://tt.amazon.com/0155095586) was from COPS, and they did recommend to validate user provided color. 

I added `options` sentence in `MyColorIsIntentHandler`, I think it makes more sense to add it in `LaunchRequest` as well.

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java (hunk 1 above)
Comment (Chris-Liao) [reply samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java]: @tianrenz, My bad. I used the flag in a confusing way as it used to be. 

I should use `boolean noAnswerProvided`, and it works like `        if(noAnswerProvided) {
            responseBuilder.withShouldEndSession(false);
        }`, and guides user to provide a valid color. Will fix in new revision.

Also, I'm not moving `isAskResponse` from `MyColorIsIntentHandler` to `WhatsMyColorIntentHandler`, I'm trying to fix couple issues.

Take a look [here](https://github.com/alexa/alexa-skills-kit-sdk-for-java/blob/2.0.x/samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/MyColorIsIntentHandler.java).
`repromptText` never gets return when `isAskResponse == false`, and `isAskResponse` will be set to `true` when user is not responding correctly. 

Basically in `MyColorIsIntentHandler `, session should keep **open** since response is asking user to do something.

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java [hunk 3]
```diff
@@ -30,7 +30,8 @@ public boolean canHandle(HandlerInput input) {
...
+                "Your options are green, blue, purple, red, orange and yellow.";
```
Comment (breedloj) [inline samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java]: I don't think it is necessary to enumerate all of the possible options here. It's verbose and less intuitive since in a natural conversational sense I shouldn't have to think about this. It is also a maintainability concern since color updates in the interaction model now also need to reflected in code.

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java (hunk 1 above)
Comment (tianrenz) [reply samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java]: Cool thanks for the explanation

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java (hunk 3 above)
Comment (Chris-Liao) [reply samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/LaunchRequestHandler.java]: Make a lot sense, updating in new revision.

File: samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java [hunk 4]
```diff
@@ -35,18 +36,26 @@ public boolean canHandle(HandlerInput input) {
...
-                            + "red";
+                            + "red.";
+            noAnswerProvided = true;
+        }
+
+        ResponseBuilder responseBuilder = input.getResponseBuilder();
+        responseBuilder.withSimpleCard("ColorSession", speechText)
+                .withSpeech(speechText);
+
+        if(noAnswerProvided) {
```
Comment (tianrenz) [inline samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java]: this can be simplified as 
```
responseBuilder.withShouldEndSession(!noAnswerProvided);
```

Comment (Chris-Liao) [reply samples/colorpicker/src/com/amazon/ask/colorpicker/handlers/WhatsMyColorIntentHandler.java]: Updating in new revision, and merge commits.

**Review by breedloj — APPROVED** [review]

**Review by tianrenz — APPROVED** [review]
